src/basic/string-util.c

```diff
@@ -176,6 +176,19 @@ char *truncate_nl(char *s) {
         return s;
 }
 
+char *truncate_nl_len(char **ibuf, size_t *_isz) {
```

Comment:
Please extend `truncate_nl()`. How about the following?
```
char *truncate_nl_full(char *s, size_t *ret_len) {
        size_t n;

        assert(s);

        n = strcspn(s, NEWLINE);
        s[n] = '\0';
        if (ret_len)
                *ret_len = n;
        return s;
}
```
Then, in string-util.h
```
char *truncate_nl_full(char *s, size_t *ret_len);
static inline char *truncate_nl(char *s) {
        return truncate_nl_full(s, NULL);
}
```

Reply:
@yuwata thank you for the review! 
From my perspective name this function with prefix "_full" is not a completely good idea as in the future someone would probably want to override it as well. I suggest using "truncate_nl_len" (like in my comment above) but I like your implementation.

Reply:
The naming is not a matter, hence `truncate_nl_len()` is ok.

Reply:
it's common convention to call functions that get shortened by other helpers `_full`